applications/GeoMechanicsApplication/tests/cpp_tests/custom_workflows/test_GeoMechanicsNewtonRaphsonErosionProcessStrategy.cpp

```diff
@@ -21,97 +21,51 @@
-{
-    auto workingDirectory = "./applications/GeoMechanicsApplication/tests/test_compare_sellmeijer/"
-                            "HeightAquiferD10L30.gid";
-    auto projectFile      = "ProjectParameters.json";
-
-    auto                                           execute = KratosExecute();
-    const Kratos::KratosExecute::CriticalHeadInfo  critical_head_info(3, 4, 0.1);
-    const Kratos::KratosExecute::CallBackFunctions call_back_functions(
-        &flow_stubs::emptyLog, &flow_stubs::emptyProgress, &flow_stubs::emptyLog, &flow_stubs::emptyCancel);
+using namespace Kratos;
 
-    const int status = execute.ExecuteFlowAnalysis(workingDirectory, projectFile, critical_head_info,
-                                                   "PorousDomain.Left_head", call_back_functions);
-
-    KRATOS_EXPECT_EQ(status, 0);
-}
-
-KRATOS_TEST_CASE_IN_SUITE(ErosionProcessStrategyTextualProgressReport, KratosGeoMechanicsIntegrationSuite)
+class ParametrizedDGeoFlowTests : public ::testing::TestWithParam<std::string>
 {
-    auto workingDirectory = "./applications/GeoMechanicsApplication/tests/test_compare_sellmeijer/"
-                            "HeightAquiferD10L30.gid";
-    auto projectFile      = "ProjectParameters.json";
-
-    auto execute = KratosExecute();
-
-    bool firstMessageFound = false;
-    bool finalMessageFound = false;
-    int  messageCount      = 0;
-
-    std::function<void(const char*)> reportTextualProgress = [&firstMessageFound, &finalMessageFound,
-                                                              &messageCount](const char* message) {
-        messageCount++;
-
-        if (strcmp(message, "Calculating head level 3m (1/12)") == 0) {
-            firstMessageFound = true;
-        }
+};
 
-        if (strcmp(message, "Calculating head level 3.8m (9/12)") == 0) {
-            finalMessageFound = true;
-        }
-    };
-
-    const Kratos::KratosExecute::CriticalHeadInfo  critical_head_info(3, 4, 0.1);
-    const Kratos::KratosExecute::CallBackFunctions call_back_functions(
-        &flow_stubs::emptyLog, &flow_stubs::emptyProgress, reportTextualProgress, &flow_stubs::emptyCancel);
-
-    const int status = execute.ExecuteFlowAnalysis(workingDirectory, projectFile, critical_head_info,
-                                                   "PorousDomain.Left_head", call_back_functions);
-
-    KRATOS_EXPECT_EQ(status, 0);
-    KRATOS_EXPECT_EQ(firstMessageFound, true);
-    KRATOS_EXPECT_EQ(finalMessageFound, true);
-    KRATOS_EXPECT_EQ(messageCount, 9);
-}
-
-KRATOS_TEST_CASE_IN_SUITE(ErosionProcessStrategyProgressReport, KratosGeoMechanicsIntegrationSuite)
+TEST_P(ParametrizedDGeoFlowTests, ErosionProcessStrategy)
 {
-    auto workingDirectory = "./applications/GeoMechanicsApplication/tests/test_compare_sellmeijer/"
-                            "HeightAquiferD10L30.gid";
-    auto projectFile      = "ProjectParameters.json";
+    // Arrange
+    auto workingDirectory =
```

Comment:
Now that we have started to clean up these tests we might as well make sure that we follow the Kratos Style Guide when it comes to naming:
```suggestion
    auto working_directory =
```

You may want to check out the other variables in this test file too.

Reply:
Good point, I think I got them all